src/textual/drivers/linux_driver.py

```diff
@@ -265,6 +266,13 @@ def more_data() -> bool:
                         for event in feed(unicode_data):
                             self.process_event(event)
         except Exception as error:
-            log(error)
+            self._app.exit(
+                return_code=1,
+                message=rich.traceback.Traceback.from_exception(
```

Comment:
I think it makes more sense to call `app.panic` here. But bear in mind that neither `exit` or `panic` are threadsafe. It's probably wise to use `app.call_later` to ensure that `panic` is called in the main thread.  

BTW I don't think you'll need all those parameters to `Traceback`. If you construct `Traceback()` in a handler it will detect the traceback automatically.